### PR TITLE
added examples folder

### PR DESCRIPTION
**Pull request by wouterdb**
added some examples 

No mechanism provided to run them yet

### CODE REVIEW

File: examples/readme.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+| OS_AUTH_URL       | The Openstack Auth URL  |
+| OS_USERNAME       | Your Openstack username |
+| OS_PASSWORD       | Your Openstack password |
+| OS_PROJECT_NAME   | The name of the project to deploy the VM to|
+
+
+2. Create a project to run the example from
+
+```bash
+cookiecutter gh:inmanta/inmanta-project-template
```
Comment (bartv) [inline examples/readme.md]: Should at least link to cookiecutter

File: examples/tenant.cf [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    admin_role = openstack::Role(role="admin",
+                    provider=admin_provider,
+                    project=project,
+                    user=admin_user,
+                    purge_on_delete=false,
+                    purged=false, # don't delete, this would lock out the admin user
+                    provides=self.provides)
+    
+    #sequence delete
```
Comment (bartv) [inline examples/tenant.cf]: Does this work with the dependency handler, maybe this logic should even go in the dep handler? Also, the comment should be a bit more verbose then "sequence delete". For example: Set dependencies between resources correctly when the resources are being purged.

File: model/_init.cf [hunk 3]
```diff
@@ -120,9 +120,14 @@ Role.project [1] -- Project.roles [0:]
 Role.user [1] -- User.roles [0:]
 
 implementation roleImpl for Role:
-    self.requires = self.project
-    self.requires = self.user
-
+    if not self.purged:
```
Comment (bartv) [inline model/_init.cf]: dep handler?

File: examples/readme.md (hunk 1 above)
Comment (wouterdb) [reply examples/readme.md]: done

File: examples/tenant.cf (hunk 2 above)
Comment (wouterdb) [reply examples/tenant.cf]: add comment

File: model/_init.cf (hunk 3 above)
Comment (wouterdb) [reply model/_init.cf]: if I can find the time, otherwise I will split this of into another ticket

Comment (wouterdb) [reply model/_init.cf]: I will split this of into another ticket

Comment (wouterdb) [reply model/_init.cf]: see #43

**Comment by wouterdb** [comment]
merged by hand

**Pull request closed without merging**